iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -442,13 +440,17 @@ public override async Task CloseAsync(CancellationToken cancellationToken)
             if (Logging.IsEnabled)
                 Logging.Enter(this, cancellationToken, nameof(CloseAsync));
 
-            _isOpen = false;
+            if (!_isOpen)
+            {
+                // Already closed so gracefully exit, instead of throw.
+                return;
+            }
 
             await _handlerSemaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
 
             try
             {
-                if (!_wasOpenCalled)
```

Comment:
We were tracking whether open was called, only to customize the error message, which seemed overly complicated (Abhipsa agreed), so we got rid of the notion.

Reply:
See old line 477